src/core/comm/coll.cc

```diff
@@ -270,16 +264,9 @@ int collInit(int argc, char* argv[])
   CHECK_MPI(MPI_Comm_get_attr(MPI_COMM_WORLD, MPI_TAG_UB, &tag_ub, &flag));
   assert(flag);
   mpi_tag_ub = *tag_ub;
-  // create mpi comms
-  mpi_comms.resize(MAX_NB_COMMS, MPI_COMM_NULL);
-  for (int i = 0; i < MAX_NB_COMMS; i++) { CHECK_MPI(MPI_Comm_dup(MPI_COMM_WORLD, &mpi_comms[i])); }
+  mpi_comms.clear();
 #else
-  thread_comms.resize(MAX_NB_COMMS);
-  for (int i = 0; i < MAX_NB_COMMS; i++) {
-    thread_comms[i].ready_flag = false;
-    thread_comms[i].buffers    = nullptr;
-    thread_comms[i].displs     = nullptr;
-  }
+  thread_comms.clear();
```

Comment:
same comment here. why is this necessary?

Reply:
fixed.